dartdoc_test/lib/src/resource.dart

```diff
@@ -1,15 +1,61 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 import 'dart:io';
 
 import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
 import 'package:analyzer/file_system/physical_file_system.dart';
 import 'package:analyzer/file_system/overlay_file_system.dart';
+import 'package:path/path.dart' as p;
+
+import 'extractor.dart';
 
 final resourceProvider =
     OverlayResourceProvider(PhysicalResourceProvider.INSTANCE);
 
 final _current = Directory.current;
 
-final currentContext = AnalysisContextCollection(
+// Directory to store generated code samples
+final testDirectory = Directory('${_current.path}/.dartdoc_test');
+
+final _contextCollection = AnalysisContextCollection(
   resourceProvider: resourceProvider,
   includedPaths: [_current.absolute.path],
-).contextFor(_current.absolute.path);
+);
+
+final currentContext = _contextCollection.contextFor(_current.absolute.path);
```

Comment:
I'd avoid making stateful global variables like `currentContext`, `testDirectory`, `resourceProvider`.

Consider making a `DartDocTestContext` and placing all the state you need on that. Or something like that. Could also be called a `TestContext` just to keep name short and simple :D

Reply:
I created a singleton class called `TestContext`.